### PR TITLE
Replace output_dir with dir and deprecate output_dir

### LINKED ISSUE #297: Deprecate `output_dir` in favor of `dir`
Per the discussion over at https://github.com/stan-dev/cmdstanpy/issues/289, for consistency we should change the `output_dir` argument of the model fitting methods to `dir` to match the argument name used in `install_cmdstan()` and the `compile()` method.

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Summary

Fixes #297 

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting 
(this will be you or your assignee, such as a university or company): 
Rok Češnovar, Uni. of Ljubljana

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
I tried to also rebuild the .html files but `pkgdown::build_reference()` made a bunch of unrelated changes so I am assuming I should have called it differently.

**Comment by jgabry** [comment]
> I tried to also rebuild the .html files but `pkgdown::build_reference()` made a bunch of unrelated changes so I am assuming I should have called it differently.

That might just be because we've merged other doc changes without regenerating the html, so they're getting included now.

File: R/args.R [hunk 1]
```diff
@@ -34,9 +34,16 @@ CmdStanArgs <- R6::R6Class(
...
-
+                          dir = NULL,
+                          validate_csv = TRUE,
+                          # deprecated
+                          output_dir = NULL) {
+      # temporary deprecation warnings
+      if (!is.null(output_dir)) {
+        warning("'output_dir' is deprecated. Please use 'dir' instead.")
+        dir <- output_dir
+      }
```
Comment (jgabry) [inline R/args.R:45]: Do we need this in args.R if we also have it in each of the mode fitting methods? Is there a  case where we only throw this warning here without also throwing it from, e.g., `$sample()`?

**Comment by jgabry** [comment]
@rok-cesnovar This looks good, just one question above. 

@mitzimorris Just wanted to double check that you're also planning on doing this in CmdStanPy. I think we agreed to just use `dir` everywhere, right?

File: R/args.R (hunk 1 above)
Comment (rok-cesnovar) [reply R/args.R:45]: Oh right. Let me double check.

Comment (rok-cesnovar) [reply R/args.R]: Actually yeah, I just removed it. cmdstanArgs is internal. No need to keep it.

**Comment by rok-cesnovar** [comment]
> That might just be because we've merged other doc changes without regenerating the html, so they're getting included now.

see commit https://github.com/stan-dev/cmdstanr/pull/312/commits/7bb23f72bd7e4cdf7b485c5fed548fef0619de81 for the changes after pkgdown::build_reference()

Thanks!

**Comment by jgabry** [comment]
> see commit [7bb23f7](https://github.com/stan-dev/cmdstanr/commit/7bb23f72bd7e4cdf7b485c5fed548fef0619de81) for the changes after pkgdown::build_reference()

Yeah looks like stuff that hadn't been updated previously. Should be fine.

**Comment by jgabry** [comment]
@mitzimorris Before I merge this, just want to confirm that this is the plan for CmdStanPy too. We decided to use `dir` everywhere right?

**Comment by mitzimorris** [comment]
overall, I don't see a huge benefit from changing this now - perhaps we should have gone with the simplest possible argument name - we didn't.

reasons not to do this:

- the name `output_dir` comes from current CmdStan argument name `output file=...` and when we flatten CmdStan's arguments, it will be `output_file` and `data_file`

- keeping `output_dir` allows for the possibility of `data_dir` later.

**Comment by jgabry** [comment]
Thanks for the comments: 

> * keeping `output_dir` allows for the possibility of `data_dir` later.

Yeah that's a good point that I hadn't considered before we talked about this. It does seem fine to have `output_dir` instead of `dir` if we're going to potentially have multiple `dir` related arguments.

**Comment by rok-cesnovar** [comment]
Thanks! Closing this and the issue then.

**Pull request closed without merging**